tests/integrations/litestar/test_litestar.py

```diff
@@ -0,0 +1,326 @@
+    sentry_init(
+        traces_sample_rate=1.0,
+        integrations=[LitestarIntegration()],
+    )
+
+    logging_config = LoggingMiddlewareConfig()
+    session_config = ServerSideSessionConfig()
+    rate_limit_config = RateLimitConfig(rate_limit=("hour", 5))
+
+    litestar_app = litestar_app_factory(
+        middleware=[
+            session_config.middleware,
+            logging_config.middleware,
+            rate_limit_config.middleware,
+        ]
+    )
+    events = capture_events()
+
+    client = TestClient(
+        litestar_app, raise_server_exceptions=False, base_url="http://testserver.local"
+    )
+    try:
+        client.get("/message")
+    except Exception:
+        pass
+
+    (_, transaction_event) = events
+
+    expected = ["SessionMiddleware", "LoggingMiddleware", "RateLimitMiddleware"]
+
+    idx = 0
+    for span in transaction_event["spans"]:
+        if span["op"] == "middleware.litestar":
+            assert span["description"] == expected[idx]
+            assert span["tags"]["litestar.middleware_name"] == expected[idx]
+            idx += 1
+
+
+def test_middleware_callback_spans(sentry_init, capture_events):
+    sentry_init(
+        traces_sample_rate=1.0,
+        integrations=[LitestarIntegration()],
+    )
+    litestar_app = litestar_app_factory(middleware=[SampleMiddleware])
+    events = capture_events()
+
+    client = TestClient(litestar_app, raise_server_exceptions=False)
+    try:
+        client.get("/message")
+    except Exception:
+        pass
+
+    (_, transaction_event) = events
+
+    expected = [
+        {
+            "op": "middleware.litestar",
+            "description": "SampleMiddleware",
+            "tags": {"litestar.middleware_name": "SampleMiddleware"},
+        },
+        {
+            "op": "middleware.litestar.send",
+            "description": "SentryAsgiMiddleware._run_app.<locals>._sentry_wrapped_send",
+            "tags": {"litestar.middleware_name": "SampleMiddleware"},
+        },
+        {
+            "op": "middleware.litestar.send",
+            "description": "SentryAsgiMiddleware._run_app.<locals>._sentry_wrapped_send",
+            "tags": {"litestar.middleware_name": "SampleMiddleware"},
+        },
+    ]
+    for idx, span in enumerate(transaction_event["spans"]):
+        assert span["op"] == expected[idx]["op"]
+        assert span["description"] == expected[idx]["description"]
+        assert span["tags"] == expected[idx]["tags"]
+
+
+def test_middleware_receive_send(sentry_init, capture_events):
+    sentry_init(
+        traces_sample_rate=1.0,
+        integrations=[LitestarIntegration()],
+    )
+    litestar_app = litestar_app_factory(middleware=[SampleReceiveSendMiddleware])
+
+    client = TestClient(litestar_app, raise_server_exceptions=False)
+    try:
+        # NOTE: the assert statements checking
+        # for correct behaviour are in `SampleReceiveSendMiddleware`!
+        client.get("/message")
+    except Exception:
```

Comment:
If there is a narrower exception we can catch here, that would be preferable. We should only catch the actual exception we expect here, any other exceptions likely should fail the test.

Reply:
I am not sure why most of the try/catch blocks (including this one) were in the original test_starlite.py tests since it does not seem the tests expect any particular exception. Maybe it was just a copy/paste from the one original test that is intentionally catching exceptions to verify exception behavior for the integration. In any case, I removed them and the tests run fine.